src/Lachain.Storage/Trie/NodeRepository.cs

```diff
@@ -44,5 +46,35 @@ public void SaveBatch(WriteBatch batch)
         {
             _rocksDbContext.SaveBatch(batch);
         }
+
+        public bool NodeIdExist(ulong id, IDbShrinkRepository _repo)
```

Comment:
I think these methods should not be a part of NodeRepository. Let's isolate these specific methods to some other object or (maybe in DbShrinkRepo). What's your opinion about it ?

Reply:
DbShrinkRepo would be a good option.